### PR TITLE
boot,dirs,image: introduce boot.MakeBootable, use it in image instead of ad hoc code

### PR DESCRIPTION
**Pull request by pedronis**
This also means that only boot refers to bootloader, and all use and referecences to snappy boot env variables (outside of tests) are in boot as well.

This stops as well gadget downloading being a one off by moving its unpacking close to the point of use.

It adds some tests about areas that weren't tested in image tests to double check the reorg.

Adds some helpersto dirs to compute some snapd directories with a non-default root, I expect this kind of helpers for just a few more dirs involved at image creation and not just runtime.

### CODE REVIEW

File: boot/boot.go [hunk 1]
```diff
@@ -273,3 +276,92 @@ func MarkBootSuccessful() error {
 
 	return bl.SetBootVars(m)
 }
+
+// MakeBootable sets up the image filesystem with the given rootdir
```
Comment (mvo5) [inline boot/boot.go]: :heart:

File: image/image_test.go [hunk 2]
```diff
@@ -884,12 +848,13 @@ func (s *imageSuite) TestSetupSeedWithBase(c *C) {
 	model := s.Brands.Model("my-brand", "my-model", map[string]interface{}{
 		"architecture":   "amd64",
 		"gadget":         "pc18",
-		"kernel":         "pc-kernel",
+		"kernel":         "pc-kernel=18",
```
Comment (pedronis) [inline image/image_test.go]: I should probably undo this change, this is actually tested in TestSetupSeedWithKernelAndGadgetTrack

File: boot/boot.go [hunk 3]
```diff
@@ -273,3 +276,92 @@ func MarkBootSuccessful() error {
...
+		"snap_try_kernel": "",
+	}
+	if model.DisplayName() != "" {
+		m["snap_menuentry"] = model.DisplayName()
+	}
+
+	for fn, info := range bootWith {
+		bootvar := ""
+
+		switch info.GetType() {
```
Comment (bboozzoo) [inline boot/boot.go]: Should this error our on TypeApp ?

Comment (pedronis) [reply boot/boot.go]: it is also not checking that there is exactly one kernel and one base mmh

I'll land and iterate
